include/envoy/server/filter_config.h

```diff
@@ -311,6 +323,16 @@ class NamedHttpFilterConfigFactory {
     return nullptr;
   }
 
+  // ProtocolOptionsConsumer
+  Upstream::ProtocolOptionsConfigConstSharedPtr
+  createProtocolOptionsConfig(const Protobuf::Message& config) override {
```

Comment:
What do you think about moving the default implementations (here and from the NamedNetworkFilterConfigFactory) to ProtocolOptionsConsumer?

Reply:
sounds good